layers/layer_wrappers.py

```diff
@@ -0,0 +1,523 @@
+
+    For example,
+
+        ## assume the input is 10000 dim
+        embed = pprl.layers.embedding(size=100)
+        ## assume the input is 100 dim
+        softmax = pprl.layers.fc(size=10000)
+        pprl.layers.paras_binding(embed, softmax)
+
+    If the two layers have different numbers of parameters, then an error will occur
+    when executing the target_layer. The user is responsible for the result of the binding.
+
+    If the src_layer has the param or bias missing in the target_layer, then the target_layer
+    still uses its own copy of parameters;
+    if the target_layer has the param or bias missing in the src_layer, then it does not matter.
+    """
+    assert LayerFunc.check_type(src_layer)
+    assert LayerFunc.check_type(target_layer)
+
+    if src_layer.param_name is not None:
+        target_layer.param_name = src_layer.param_name
+    elif target_layer.param_name is not None:
+        warnings.warn(target_layer.__class__ + "'s params are not bound to " + src_layer.__class__)
+
+    if src_layer.bias_name is not None:
+        target_layer.bias_name = src_layer.bias_name
+    elif target_layer.param_name is not None:
+        warnings.warn(target_layer.__class__ + "'s biases are not bound to " + src_layer.__class__)
+
+
+def create_parameter(shape, dtype):
+    """
+    Return a function that creates paddle.fluid.layers.create_parameter
+    with shape and dtype.
+    """
+    param_name = "para_%d.w" % LayerCounter.create_parameter
+    LayerCounter.create_parameter += 1
+
+    class CreateParameter_(LayerFunc):
+        def __init__(self):
+            super(CreateParameter_, self).__init__(param_name)
+
+        def __call__(self,
+                     name=None,
+                     attr=None,
+                     is_bias=False,
+                     default_initializer=None):
+            attr = check_param_name(attr)
+            if attr:
+                attr.name = self.param_name
+
+            return layers.create_parameter(
+                shape=shape,
+                dtype=dtype,
+                name=name,
+                attr=attr, # this cannot be None
+                is_bias=is_bias,
+                default_initializer=default_initializer)
+
+    return CreateParameter_()
+
+
+def fc(size):
+    """
+    Return a function that creates a paddle.fluid.layers.fc layer with the
+    specified output size.
+    """
+    param_name = "fc_%d.w" % LayerCounter.fc
+    bias_name = "fc_%d.wbias" % LayerCounter.fc
+    LayerCounter.fc += 1
+
+    class FC_(LayerFunc):
+        def __init__(self):
+            super(FC_, self).__init__(param_name, bias_name)
+
+        def __call__(self,
+                     input,
+                     num_flatten_dims=1,
+                     param_attr=None,
+                     bias_attr=None,
+                     use_mkldnn=False,
+                     act=None,
+                     is_test=False,
+                     name=None):
```

Comment:
I feel most arguments except "input" and "is_test" should be set at __init__()

Reply:
That might not be as flexible as now. Suppose there are two fc layers and one of them has a linear act and one of them has a sigmoid act, then the parameters cannot be shared.

Reply:
And if one fc layer has num_flatten_dims=1 while the other has num_flatten_dims=2, then their parameters cannot be shared automatically. Unless each time we manually bind their parameters by providing one's parameters to the other's init().